src/transaction/external-executor.ts

```diff
@@ -186,12 +188,26 @@ export class ExternalTransactionExecutor implements ITransactionExecutor {
               })
         ).signedTx;
 
-        signedTxGlobal = signedTx;
+        let processedSignedTx = signedTx;
+
+        //Check if the signed transaction comes from the browser wallet
+        if ('signedData' in signedTx) {
+          const parsedSignedData = JSON.parse(signedTx.signedData);
+
+          processedSignedTx = {
```

Comment:
So, this would go to the signer.
It will stay consistent with the others and make this function cleaner (it requires some cleaning anyway).